### PR TITLE
Seperates map reader parsing from loading

### PR DESCRIPTION
**Pull request by Cyberboss**
Renamed /dmm_suite to /datum/maploader, load_map now returns a /datum/parsed_map no other functional changes

This is mainly just to make the code more sane to work with in the future

### CODE REVIEW

**Comment by SpaceManiac** [comment]
code/modules/mapping/reader.dm:270:error: &&: missing comma ',' or right-paren ')'

**Comment by SpaceManiac** [comment]
```
code/modules/mapping/reader.dm:147:error: grid_models: undefined var
code/controllers/subsystem/mapping.dm:179:error: parsed_map.bounds: undefined var
```

**Comment by SpaceManiac** [comment]
lots of these
```
[23:58:28] Runtime in reader.dm,187: Undefined model key in DMM.
  proc name: load map impl (/datum/maploader/proc/load_map_impl)
  src: /datum/maploader (/datum/maploader)
  call stack:
  /datum/maploader (/datum/maploader): load map impl(_maps/map_files/BoxStation/Box..., 1, 1, 2, null, null, 1, -1e+31, 1e+31, -1e+31, 1e+31, 0)
  /datum/maploader (/datum/maploader): load map(_maps/map_files/BoxStation/Box..., 0, 0, 2, null, null, 1, null, null, null, null, null)
  Mapping (/datum/controller/subsystem/mapping): LoadGroup(/list (/list), "Station", "map_files/BoxStation", /list (/list), /list (/list), /list (/list), 0)
  Mapping (/datum/controller/subsystem/mapping): loadWorld()
  Mapping (/datum/controller/subsystem/mapping): Initialize(863054)
  Master (/datum/controller/master): Initialize(10, 0)
```

File: code/modules/mapping/dmm_suite.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-dmm_suite{
+/datum/maploader{
```
Comment (AnturK) [inline code/modules/mapping/dmm_suite.dm]: If you're changing this file, throw away {}'s and relative pathing.

File: code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -175,7 +175,8 @@ SUBSYSTEM_DEF(mapping)
 	var/total_z = 0
 	for (var/file in files)
 		var/full_path = "_maps/[path]/[file]"
-		var/bounds = loader.load_map(file(full_path), 1, 1, 1, cropMap=FALSE, measureOnly=TRUE)
+		var/datum/parsed_map/pm = loader.load_map(file(full_path), 1, 1, 1, cropMap=FALSE, measureOnly=TRUE)
+		var/bounds = pm?.bounds
```
Comment (AnturK) [inline code/controllers/subsystem/mapping.dm]: Why null-check if it will runtime two lines down anyway.

**Review by AnturK — COMMENTED** [review]
Looks alright, just some nitpicks.

File: code/modules/mapping/dmm_suite.dm (hunk 1 above)
Comment (SpaceManiac) [reply code/modules/mapping/dmm_suite.dm]: do this as a code cleanup in a later PR or the diff will be unreadable

**Pull request merged**